compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -185,15 +185,7 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
         }
         let concrete = infcx.const_eval_resolve(param_env, uv.expand(), Some(span));
         match concrete {
-            Err(ErrorHandled::TooGeneric) => Err(if uv.has_infer_types_or_consts() {
-                NotConstEvaluatable::MentionsInfer
-            } else if uv.has_param_types_or_consts() {
-                infcx
-                    .tcx
-                    .sess
-                    .delay_span_bug(span, &format!("unexpected `TooGeneric` for {:?}", uv));
-                NotConstEvaluatable::MentionsParam
-            } else {
+            Err(ErrorHandled::TooGeneric) => Err({
```

Comment:
can you move the `Err` into the block or directly use the `guar` in `NotConstEvaluatable::Error` without first assigning it to a local variable?